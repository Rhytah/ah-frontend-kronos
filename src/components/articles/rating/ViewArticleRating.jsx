/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/mouse-events-have-key-events */
import React from 'react';
import { PropTypes } from 'prop-types';
import '../../../assets/scss/Landing.scss';


const isCheckedClass = (position, rating) => {
  let ratingClass = 'fa fa-star rating-star';

  ratingClass = position <= rating ? ratingClass.concat(' checked') : ratingClass;
  return ratingClass;
};
const ViewArticleRating = ({
  rating, handleRating, handleMouseOver, handleMouseOut,
}) => {
  const ratingslist = [1, 2, 3, 4, 5];
  return ratingslist.map((theRating, index) => (
    <span
      // eslint-disable-next-line react/no-array-index-key
      key={index}
      onClick={() => handleRating(theRating)}
      onMouseOver={() => handleMouseOver(theRating)}
      onMouseOut={handleMouseOut}
      className={isCheckedClass(theRating, rating)}
    />
  ));
};

export default ViewArticleRating;

ViewArticleRating.propTypes = {
  rating: PropTypes.number,
  handleRating: PropTypes.func,
  handleMouseOver: PropTypes.func,
  handleMouseOut: PropTypes.func,
};

ViewArticleRating.defaultProps = {
  rating: 0,
  handleRating: () => {},
  handleMouseOver: () => {},
  handleMouseOut: () => {},
};
