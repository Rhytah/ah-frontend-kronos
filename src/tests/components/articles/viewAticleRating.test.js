/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import ViewArticleRating from '../../../components/articles/rating/ViewArticleRating';

describe('Component: ViewArticleRating', () => {
  test('MyComponent renders with default props', () => {
    const wrapper = shallow(<ViewArticleRating />);
    expect(wrapper).toMatchSnapshot();
  });
});
