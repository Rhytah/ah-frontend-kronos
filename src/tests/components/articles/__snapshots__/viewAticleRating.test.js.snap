// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ViewArticleRating MyComponent renders with default props 1`] = `ShallowWrapper {}`;
